### PR TITLE
Order Details: Show more button on Product section for recreating shipping labels

### LINKED ISSUE #4676: Shipping Labels: allow re-creating a shipping label for all cases (refunded, existing, etc...)
As discussed [here](https://a8c.slack.com/archives/C026HH75ECA/p1627308002046400?thread_ts=1626460086.013600&cid=C026HH75ECA), we should allow re-creating shipping labels for all cases, even if we have non-refunded existing labels, the button to re-create the label will be shown in the overflow menu, presenting the bottom action sheet. It should be added in the action sheet where we already display the action to refund a shipping label.

Relevant discussion [here](https://a8c.slack.com/archives/CGPNUU63E/p1615913529004000?thread_ts=1615285371.106700&cid=CGPNUU63E).

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes #4676 

# Description
This PR adds More button on Product section of Order Details screen to give option for recreating shipping labels. This button should only be visible if all these conditions are met:
- Order is eligible for creating shipping labels
- Order is not eligible for card present payment
- Order has at least one associating shipping label, refunded or not.

# Changes
- Updates order details data source to change header style of Product section based on whether recreating shipping labels is allowed.
- Updates order details view model and view controller to show action sheet / popover when tapping the More button on Product section.

# Screenshots
| Order with refunded label | Order with non-refunded label |
| ----- | ----- |
|  |  

| Popover on iPad |
| ----- |
| Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2021-10-21 at 11 07 35 |

# Testing
### Ineligible orders
On a store that hasn't configured WCShip plugin, make sure that More button is never visible on Product section.

### Eligible orders without shipping labels
On a store that has configured WCShip plugin, select an order that doesn't have an associating shipping label. Make sure that More button is not visible on Product section.

### Eligible orders that's also eligible for card present payment
On a store that has configured WCShip and WCPay plugins, select a cash-on-delivery order that is still waiting for payment. Make sure that More button is not visible on Product section.

### Eligible orders with shipping labels created, regardless of status for any of those labels
On a store that has configured WCShip, select an order with at least one shipping label created. Notice that More button is visible on Product section - whether the label is refunded or not.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by pmusolino — COMMENTED** [review]
I tested it on this order, but am I wrong, or should I see the button to create a new shipping label?
It should be an order with non-refunded label. WDYT?

**Comment by itsmeichigo** [comment]
@pmusolino as commented on #4676, I'm showing the option to re-create shipping labels in the More button of the Product section, not on the purchased label, so you should be able to find it in the highlighted area here:

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 7.9
 -----
 - [*] Fix: after disconnecting a site or connecting to a new site, the sites in site picker (Settings > Switch Store) should be updated accordingly. The only exception is when the newly disconnected site is the currently selected site. [https://github.com/woocommerce/woocommerce-ios/pull/5241]
-
+- [*] Order Details: Show More button on Product section to allow recreating shipping labels. [https://github.com/woocommerce/woocommerce-ios/pull/5255]
```
Comment (pmusolino) [inline RELEASE-NOTES.txt]: nit:
```suggestion
- [*] Order Details: Show "More button" on a Product detail to allow recreating shipping labels. [https://github.com/woocommerce/woocommerce-ios/pull/5255]
```

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift [hunk 2]
```diff
@@ -60,6 +60,13 @@ final class OrderDetailsDataSource: NSObject {
             !isEligibleForCardPresentPayment
     }
 
+    /// Whether the option to re-create shipping labels should be visible.
+    ///
+    var shouldAllowRecreatingShippingLabels: Bool {
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift:65]: Super clear!

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift [hunk 3]
```diff
@@ -546,6 +539,21 @@ private extension OrderDetailsViewController {
...
+            }
+        }
+        shippingLabelFormVC.onLabelSave = { [weak self] in
+            guard let self = self else { return }
+            self.navigationController?.popToViewController(self, animated: true)
+        }
+        shippingLabelFormVC.hidesBottomBarWhenPushed = true
+        navigationController?.show(shippingLabelFormVC, sender: self)
+    }
+
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift:556]: Great idea to extract it in a separate method 👍

**Review by pmusolino — APPROVED** [review]
:shipit: works like a charm. Thanks for working on this feature!

**Pull request merged**